vault/resource_userpass_user_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAccUserpassUser_basic(t *testing.T) {
+	backend := acctest.RandomWithPrefix("userpass")
+	resName := "vault_userpass_user.user"
+	username := "john_doe"
+	password := "supersecretpassword"
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccUserpassUserCheckDestroy,
```

Comment:
We don't need to specifically check if the resource is destroyed. Instead we can use an existing util to make sure the entire mount has been destroyed (thereby confirming that everything under the mount was also cleaned up). For an example of using the util to to check if the mount is destroyed, see [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_ldap_secret_backend_dynamic_role_test.go#L66) as an example